hub/core/storage/lru_cache.py

```diff
@@ -245,7 +245,7 @@ def _insert_in_cache(self, path: str, value: Union[bytes, Cachable]):
         Raises:
             ReadOnlyError: If the provider is in read-only mode.
         """
-        self.check_readonly()
+
```

Comment:
i think this removal shouldn't have any side-effects. let me know if you foresee any

Reply:
This actually makes sense. Even when you read you need to put data into the cache and having the check would have prevented that.